libraries/ms-common/src/stm32f0xx/i2c.c

```diff
@@ -0,0 +1,129 @@
+#include "i2c.h"
+#include <stdbool.h>
+#include "stm32f0xx.h"
+#include "log.h"
+
+// Arbitrary timeout
+#define I2C_TIMEOUT 1000000
+#define I2C_TIMEOUT_WHILE_FLAG(i2c, flag, status) \
+do { \
+  uint32_t timeout = (I2C_TIMEOUT); \
```

Comment:
Yes, I know it's used for the busy loop. It just seems to me that it isn't really measuring a time interval, and so ``timeout`` is a bit of a misnomer..